Release 2.2.0

scwatts opened the pull request.

scwatts committed:
Add modules for WISP workflow

scwatts committed:
Implement relevant subworkflows for WISP workflow

scwatts committed:
Add WISP workflow

scwatts committed:
Add modules for panel resource creation workflow

scwatts committed:
Input / parameter handling for WISP workflow

rhassaine committed:
Convert ESVEE to a single module

luan-n-nguyen committed:
Add options for prepare reference only mode

luan-n-nguyen committed:
Add CLI arg ref_data_types and remove arg prepare_reference_only

luan-n-nguyen committed:
Add new mode prepare_reference

luan-n-nguyen committed:
Update compute resource recommendations

scwatts committed:
Implement panel resource creation subworkflows

scwatts committed:
Add panel resource creation workflow

reecejones-hartwig committed:
Add module_log_level param to control all WiGiTS module log levels

luan-n-nguyen committed:
Add CRAM/CRAM_REDUX/CRAI filetypes. Internally, these are treated the same as BAM/BAM_REDUX/BAI files

luan-n-nguyen committed:
Make processes_manual run specified processes, and make processes_include run processes excluded by default

luan-n-nguyen committed:
Remove unused input file type REDUX_DUP_FREQ_TSV

luan-n-nguyen committed:
Update HMF tools arguments

luan-n-nguyen committed:
Implement purity estimate mode type

luan-n-nguyen committed:
Adjust workflow and subworkflow for new HMF tool arguments

luan-n-nguyen committed:
Apply WISP workflow updates and fixes

luan-n-nguyen committed:
Update arguments for tools called in WISP workflow

luan-n-nguyen committed:
Update reference data

luan-n-nguyen committed:
Run NEO by default

luan-n-nguyen committed:
Apply fixes and updates

FriederikeHanssen committed:
add reports to tower.yml

FriederikeHanssen committed:
add folder structure

luan-n-nguyen committed:
Apply logging argument to remaining HMF tools

luan-n-nguyen committed:
Provide prep config to prepare reference in purity estimate and panel resource creation workflows

FriederikeHanssen committed:
Apply suggestions from code review

Co-authored-by: Stephen Watts <scwatts@users.noreply.github.com>

luan-n-nguyen committed:
When determining prepare_reference output dir, get git branch directly from workflow.manifest.version

luan-n-nguyen committed:
In panel resource creation, remove gene-utils call to create SAGE ActionableCodingPanel and CoverageCodingPanel files

luan-n-nguyen committed:
Add option to override default driver_gene_panel in ref_data_hmf_data_path

FriederikeHanssen committed:
Merge pull request #220 from FriederikeHanssen/reporting

Add reports to tower.yml

scwatts committed:
Post release bump

luan-n-nguyen committed:
Update oncoanalyser version in docs to 2.2.0

luan-n-nguyen committed:
Update docs related to preparing and configuring ref data

luan-n-nguyen committed:
Add run mode and panel resource creation docs

scwatts committed:
Add platform-agnostic command file publishing

luan-n-nguyen committed:
Add WISP documentation

nf-core-bot committed:
Template update for nf-core/tools version 3.3.2

luan-n-nguyen committed:
Update instructions for using custom genomes

luan-n-nguyen committed:
Rename ref_data_type bwa_index to bwamem2_index

luan-n-nguyen committed:
Add docs for log output directory

luan-n-nguyen committed:
Paraphrasing and fix some typos

luan-n-nguyen committed:
Emit all .command.* files but only publish .{sh,out,err,log} files

scwatts committed:
Switch bwa-plus to bwa-mem2 2.3

luan-n-nguyen committed:
Remove old panel resource creation config

luan-n-nguyen committed:
Fix log publishing paths for normalisation steps in panel resource creation

luan-n-nguyen committed:
Publish .command.run files

luan-n-nguyen committed:
Apply fix to SAGE append run conditions and inputs

luan-n-nguyen committed:
Update HMF containers

- Bump AMBER to 4.2
- Bump BamTools to 1.4.2
- Bump CHORD to 2.1.2
- Bump COBALT to 2.1
- Bump ESVEE to 1.1.2
- Bump ISOFOX to 1.7.2
- Bump LILAC to 1.7.1
- Bump LINX to 2.1
- Bump NEO to 1.2.1
- Bump ORANGE to 4.1
- Bump PAVE to 1.8
- Bump PURPLE to 4.2
- Bump REDUX to 1.2
- Bump SAGE to 4.1
- Bump VirusInterpreter to 1.7.1

luan-n-nguyen committed:
Use WISP 1.2 Bioconda package

alexiswl committed:
Support relative paths for fastq data

scwatts committed:
Merge pull request #222 from nf-core/post-release-bump

Post release bump

scwatts committed:
Merge pull request #230 from nf-core/panel-resource-creation-workflow

Implement panel resource creation workflow

scwatts committed:
Merge branch 'nf-core/dev' into wisp-workflow

scwatts committed:
Merge pull request #231 from nf-core/wisp-workflow

Implement WISP workflow

scwatts committed:
Merge remote-tracking branch 'nf-core/dev' into improve-prepare-reference

scwatts committed:
Merge pull request #232 from nf-core/improve-prepare-reference

Improve the 'prepare reference' feature

scwatts committed:
Merge pull request #233 from nf-core/update-documentation

Update documentation for the 2.2.0 release

scwatts committed:
Merge pull request #234 from nf-core/hartwig-updates

Apply Hartwig updates, fixes

scwatts committed:
Merge remote-tracking branch 'nf-core/dev' into publish-selected-command-files

scwatts committed:
Merge pull request #235 from nf-core/publish-selected-command-files

Publish selected .command files

scwatts committed:
Merge remote-tracking branch 'nf-core/dev' into template-3.3.2-merge

scwatts committed:
Publish all SAGE outputs into separate directories

Previously SAGE append would clobber the somatic and germline outputs
under certain conditions. The change here reorganises all SAGE outputs
such that clobbering no longer can occur.

scwatts committed:
Update nf-test snap for default test

scwatts committed:
Merge pull request #223 from nf-core/nf-core-template-merge-3.3.2

Important! Template update for nf-core/tools v3.3.2

scwatts committed:
Apply hmftools log level through process config

scwatts committed:
Apply formatting and syntax changes

scwatts committed:
Refactor all Groovy code under workflow blocks

scwatts committed:
Add prepare reference workflow

scwatts committed:
Update reference data paths

scwatts committed:
Apply fixes, cleanup for ESVEE module

scwatts committed:
Apply fixes, cleanup to SAGE processes

scwatts committed:
Handle more FASTQ, longitudinal sample input cases

scwatts committed:
Reintroduce use of decoy sequence images for ESVEE

scwatts committed:
Set min stringr stringi version for CUPPA

Force install of compatible icu version 73.2, the newer 75.1 version
available in conda-forge can be installed with older the r-stringr and
r-stringi recipes but is incompatible.

scwatts committed:
Allow prepare reference through standard modes

scwatts committed:
Set and use input paths for panel resource creation

scwatts committed:
Disallow 'OFF' hmftools_log_level value

All WiGiTS tools that accept the log_level argument support the 'OFF'
value except CHORD, which causes oncoanalyser to crash. Therefore, this
log_level / hmftools_log_level value is not allowed.

scwatts committed:
Update CHANGELOG.md

scwatts committed:
Merge pull request #229 from alexiswl/bugfix/support-relative-paths-for-fastq-data

Support relative paths for fastq data

scwatts committed:
Bump to 2.2.0

scwatts committed:
Add note about CRAM requirements

scwatts committed:
Merge pull request #241 from nf-core/general-updates-and-fixes

Apply minor fixes and updates

scwatts committed:
Merge pull request #242 from nf-core/prepare-release-2.2.0

Prepare release 2.2.0

danilodileo commented on docs/usage/faq_and_troubleshooting.md:
```diff
@@ -291,32 +290,6 @@ The `work/` directory can be hard to navigate due to the `<short_hash>/<long_has
 structure, which allows you to manually find the target process directory.
 
-## Saving logs from the `work/` directory
-
```
you don't need this section anymore?

scwatts replied:
That's right! This has been implemented at the module level and documentation moved to docs/outputs.md

scwatts committed:
Fix check for multiple longitudinal sample entries

danilodileo reviewed (approved):
The pipeline looks well written and organised. I read through the files but I haven't spend too much time since I am not familiar with most of the tools implemented here.
I am testing the pipeline and I will let you know if I encounter major errors/issues.

I have just few comments:
- As I understood you only have one nf-test for stub mode. shouldn't be good to include more?
- You have many modules in local. it would be good to consider to move some of them in nf-core/modules since other developers might find it useful.
- nf-core AWS fill size tests fail. it needs to be fixed

great job!

scwatts committed:
Allow test profile to be run by nf-test

scwatts committed:
Limit nf-test GH Action to stub test

The GH Action self-hosted RunsOn instances do not have sufficient disk
space to execute non-stub tests (currently 80 GB of EBS volume according
to these docs: https://runs-on.com/configuration/job-labels/#disk). Also
using nf-test tags as providing paths to desired tests is not honoured
when --changed-since is supplied.

scwatts committed:
Update README.md with reviewer feedback

scwatts committed:
Update, fix documentation

scwatts committed:
Merge pull request #245 from nf-core/review-and-other-adjustments

Reviewer recommendations and other adjustments

scwatts commented:
Thank you for the review @danilodileo!

I've now added the `test` profile as a new nf-test case. The reason this wasn't initially done is that the GH Action automatically executes all tests, and the runner instances don't have sufficient disk space to store the reference data needed for the `test` profile (the stub run is configured in such a way that no reference data is required). In order to handle this, I've also set the corresponding GH Action to only execute the stub run; the new test can otherwise be run with `nf-test test` or similar

We're happy to explore making these tools more easily accessible to the community, though in their current state they are strongly interdependent on one another and running them in a standalone way isn't really possible, so the modules aren't individually that useful to others

I confirmed the AWS full sized test is also failing (though only on manual trigger), and this appears to be caused by some unspecified error during submission of the Seqera Platform job. Looks like it's a [known issue](https://nfcore.slack.com/archives/C04QR0T3G3H/p1756367052853309) with the current advice to ask someone from the nf-core team to trigger the job once the release is completed

scwatts commented on README.md:
```diff
@@ -5,14 +5,14 @@
   </picture>
 </h1>
 
```
Updated this, thank you!

ohofmann reviewed (approved):
I am okay with the proposed changes — thanks for all the work (and the external reviewers).
